internal/bminventory/inventory.go

```diff
@@ -340,9 +340,82 @@ func (b *bareMetalInventory) setDefaultRegisterClusterParams(ctx context.Context
 		}
 	}
 
+	params.NewClusterParams.NetworkType, err = getDefaultNetworkType(params)
+	if err != nil {
+		log.Error(err)
+		return params, err
+	}
+
 	return params, nil
 }
 
+// If the cluster is SNO, or the IP version is IPv6 (based on ClusterNetworkCidr param), or the OpenShift version >= 4.12,
+// the network type will be set to "OVNKubernetes", otherwise it will be set to "OpenShiftSDN"
+func getDefaultNetworkType(params installer.V2RegisterClusterParams) (*string, error) {
+	// If there is not a network specified already
+	if params.NewClusterParams.NetworkType == nil {
+
+		var (
+			isOpenShiftVersionRecentEnough bool
+			openShiftVersionParam          *version.Version
+		)
+
+		// If an OpenShift version is not specified, we do not assume it is >= 4.12
+		if params.NewClusterParams.OpenshiftVersion == nil {
+			isOpenShiftVersionRecentEnough = false
+		} else {
+			openShiftVersion, err := version.NewVersion(swag.StringValue(params.NewClusterParams.OpenshiftVersion))
+			openShiftVersionParam = openShiftVersion
+			if err != nil {
+				return params.NewClusterParams.NetworkType, err
+			}
+			thresholdVersion, err := version.NewVersion("4.12")
```

Comment:
Currently version library compares as follow: x.y.z > x.y.z-rc > x.y.z-fc > x.y.z-ec, so it will not catch pre-release versions of 4.12.0. If we wish to do so I can change the threshold version to 4.12.0-ec or use a different function that ignores pre-release data